### ISSUE TITLE
[Idea] Queue part of the BP for a new miner

### ISSUE THREAD

**Issue by slipher** [Unvanquished/gameplay#52]
The build point mechanics are designed to ensure gradual starvation of build points over time, so that losing a structure is a bigger and bigger problem as time goes on. However, there is a loophole with miners - they are free and can be rebuilt an unlimited number of times without consequences. So destroying it accomplishes little for the opposing team.

I propose that when a new miner is constructed, (0.5 * g_buildPointBudgetPerMiner) build points should be added to the queue.  This ensures that destroying miners in the late game will have some real consequences. But you still get up to (0.5 * g_buildPointBudgetPerMiner) BP available immediately, so you can build forwards and whatnot.

One technical issue: how to handle miners which are destroyed under construction? If we queue the BP only upon completion, this creates a perverse incentive that the opposing team should try to wait for construction to finish before destroying a miner. But if we queue the BP immediately on placement, it means a temporary loss of BP while the miner is under construction. But maybe the latter makes sense: it's as if the miner has a BP cost, but you're allowed to go into "debt" to build it unlike other structures.

**Comment by necessarily-equal** [comment]
So basically https://github.com/Unvanquished/Unvanquished/pull/1552 but with a 1/2 factor ?

**Comment by slipher** [comment]
> So basically [Unvanquished/Unvanquished#1552](https://github.com/Unvanquished/Unvanquished/pull/1552) but with a 1/2 factor ?

Yeah, I think when that was posted I thought that it was too complicated and suggested using the queueing mechanism instead.

**Comment by slipher** [comment]
Also #1552 doesn't do anything to make rebuilding more difficult as the game time increases.

**Comment by DolceTriade** [comment]
Seems reasonable and trivial to experiment with.

**Comment by bmorel** [comment]
Agree on the idea.
